design/specs/visual-query-builder.md

```diff
@@ -0,0 +1,236 @@
+Wireframes
+
+[Column Sources](https://share.balsamiq.com/c/ewCxvkjS8VoScKfgb9hFos.png)
+
+### 2.3 Adding Formulas
+
+Formulas are used to generate new columns based on different parameters. To access the list of formulas, the user start the `Add Column` process and selects the option `From Formula` at the top of the inspector panel. Selecting a formula will open a form that users can fill out to determine the values of the new column.
+
+Depending on the selected formula, different settings will be available.
+More on formulas and specific details for each will be covered on a separate issue.
+
+Columns generated from formulas will display a formula icon indicator in the column header.
+
+---
+Wireframes
+
+[Added Formula](https://share.balsamiq.com/c/vWfJ9sUYWJxed5Zo5WtPGi.png)
+
+### 2.4 Filtering Input Column Values
+
+Columns that link to multiple records can have filters applied to them to retrieve only values that match user-specified criteria. Multiple filters are allowed for each input column.
+
+Filter options will be determined by the data type of the input column.
+
+Filters can be added by clicking on the `Add Filter` option from the query configuration panel or directly from the results table column header menu.
+
+Columns with filters applied will display a filter icon indicator in the column header.
+
+---
+Wireframes
+
+[Added Input Filter](https://share.balsamiq.com/c/tK2hZy6FB4zVYpN56ejpBk.png)
+
+### 2.5 Aggregating Input Column Values
+
+In cases where a link may refer to multiple values, input columns can be aggregated. In such cases, the system will perform automatic aggregation based on the data type of the referenced column. Users can change the type of aggregation at any time.
+
+Aggregated columns will have an aggregation icon indicator in the column header.
+
+---
+Wireframes
+
+[Added Input Aggregation](https://share.balsamiq.com/c/fNcbvQ1q5xMyni5JxBg3Rc.png)
+
+### 2.6 Applying a Formula to an Input Column
+
+Direct input columns can also be transformed into formulas. The available formulas will depend on the input column type.
+
+To transform the input column into a formula, the user will select the `Apply Formula` option in the column header menu.
+
+---
+Wireframes
+
+[Applying a Formula](https://share.balsamiq.com/c/nEzYaNKNTSv9EFwzwtfSof.png)
+
+## 3. Applying Transformations to the Result Table
+
+The result table refers to the resulting table from all input columns selected and added, including filters and aggregations. Transforming this output can be done by applying output filters, sorts, and summarizations to the result table.
+
+### 3.1. Filtering the Output Table
+
+Users can filter the output table by adding a filter step to any column from the result table. The column selector, in this scenario, will only allow users to select input columns rather than the whole column list from the column selection stage.
+
+---
+Wireframes
+
+[Output Filter](https://share.balsamiq.com/c/dAVYnp8VnG2r2HzkSZ3rgp.png)
```

Comment:
I find using the text `Filter` for both input and output to be confusing. Can we rename this to be more descriptive or add additional help content in both places describing what they do?

Reply:
Agreed. See my reply to Sean's same question below:
> Rather than calling it filter, we could call it filter records or include only records that match this condition.